tm2/pkg/bft/rpc/core/routes.go

```diff
@@ -8,16 +8,15 @@ import (
 // NOTE: Amino is registered in rpc/core/types/codec.go.
 var Routes = map[string]*rpc.RPCFunc{
 	// info API
-	"health":        rpc.NewRPCFunc(Health, ""),
-	"status":        rpc.NewRPCFunc(Status, ""),
-	"net_info":      rpc.NewRPCFunc(NetInfo, ""),
-	"blockchain":    rpc.NewRPCFunc(BlockchainInfo, "minHeight,maxHeight"),
-	"genesis":       rpc.NewRPCFunc(Genesis, ""),
-	"block":         rpc.NewRPCFunc(Block, "height"),
-	"block_results": rpc.NewRPCFunc(BlockResults, "height"),
-	"commit":        rpc.NewRPCFunc(Commit, "height"),
-	//"tx":                   rpc.NewRPCFunc(Tx, "hash,prove"),
-	//"tx_search":            rpc.NewRPCFunc(TxSearch, "query,prove,page,per_page"),
```

Comment:
just out of curiousity, are you planning to add back tx_search? if not, why not? same goes for prove parameter. mostly asking for my own tm2-learning :)

Reply:
If I'm reading the ancient papyrus code, this `tx_search` endpoint was basically a "filter" for transactions.
For example, txs from a specific address, above a certain amount...

We are adding this kind of functionality to our `tx-indexer`, with even more abilities. As far as our discussions go with adding this sort of thing out of the box in tm2, currently it's been blocked after discussions in #546

Reply:
@zivkovicmilos If I understood correctly both `tx` (in this PR) and `tx-indexer` will  provide tx search functionality while indexer being more flexible.

My question is...
when multi node is available and certain node is synced from 1234 block not from genesis and running its own tx-indexer, can we search transaction that happened before 1234 block ?

Reply:
@r3v4s No, we're not adding tx searching, because the tx indexer provides this already. This PR just adds a way to serve TX results from RPC, but no search mechanism.

Also, what you're talking about refers to _blocks_, not _transactions_. For this, there are already the RPC endpoints `block`, and I believe `blockchain` does what you're looking for :)

Hope that helps.